src/main/java/org/wildfly/security/sasl/scram/ScramUtil.java

```diff
@@ -217,10 +217,10 @@ public static int decimalDigitCount(int num) {
             StringPrep.encode(password, b, StringPrep.PROFILE_SASL_QUERY);
             mac.init(new SecretKeySpec(b.toArray(), mac.getAlgorithm()));
             mac.update(salt, saltOffs, saltLen);
-            mac.update((byte) 1);
             mac.update((byte) 0);
             mac.update((byte) 0);
             mac.update((byte) 0);
+            mac.update((byte) 1);
```

Comment:
Yes, because only in such case give it same hash as is in example in RFC:
https://tools.ietf.org/html/rfc5802#section-5